### ISSUE TITLE
Random numbers drawn after instantiating GPT2 model differ between `adapter-transformers` v3.1.0 and `transformers` v4.23.1

### ISSUE THREAD

**Issue by couturierc** [adapter-hub/adapter-transformers#484]
## Environment info
- `adapter-transformers` version: 3.1.0
- Platform: Linux WSL2-x86_64-with-glibc2.31
- Python version: 3.9.2
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 1.12.0+cu102 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

## Information

Model I am using (Bert, XLNet ...): GPT2

Language I am using the model on (English, Chinese ...): English

Adapter setup I am using (if any): None

The problem arises when using:
* [x] my own modified scripts: see script below

The tasks I am working on:
* [x] this applies to any task where extra, randomly initialized, layers (e.g. linear layers) are added on top of an imported model.

## To reproduce

Code to reproduce the behavior:

```
# initial checks: python version, transformers version, adapter-transformers version
import sys
import pkg_resources

print("python version:", sys.version)

try:
    pkg_resources.get_distribution("transformers").version
except pkg_resources.DistributionNotFound as err:
    print("transformers module not found")
else:
    print(
        f"transformers=={pkg_resources.get_distribution('transformers').version}",
    )

try:
    pkg_resources.get_distribution("adapter-transformers").version
except pkg_resources.DistributionNotFound as err:
    print("adapter-transformers module not found")
else:
    print(
        f"adapter-transformers=={pkg_resources.get_distribution('adapter-transformers').version}",
    )

# import modules
from transformers import AutoTokenizer, AutoModel
import torch

torch.manual_seed(42)
# generate some random input with torch
print(torch.randint(0, 5, (1, 10)))
# adapter-transformers==3.1.0: tensor([[2, 2, 1, 4, 1, 0, 0, 4, 0, 3]])
# adapter-transformers==3.0.0: tensor([[2, 2, 1, 4, 1, 0, 0, 4, 0, 3]])
# transformers==4.23.1:        tensor([[2, 2, 1, 4, 1, 0, 0, 4, 0, 3]])

tokenizer = AutoTokenizer.from_pretrained("gpt2")
print(torch.randint(0, 5, (1, 10)))
# adapter-transformers==3.1.0: tensor([[3, 4, 0, 4, 1, 2, 0, 0, 2, 1]])
# adapter-transformers==3.0.0: tensor([[3, 4, 0, 4, 1, 2, 0, 0, 2, 1]])
# transformers==4.23.1:        tensor([[3, 4, 0, 4, 1, 2, 0, 0, 2, 1]])

model = AutoModel.from_pretrained("gpt2")
print(torch.randint(0, 5, (1, 10)))
# adapter-transformers==3.1.0: tensor([[0, 1, 1, 3, 4, 1, 2, 4, 2, 3]])
# adapter-transformers==3.0.0: tensor([[0, 4, 3, 3, 2, 2, 3, 3, 2, 1]])
# transformers==4.23.1:        tensor([[0, 4, 3, 3, 2, 2, 3, 3, 2, 1]])
```

## Expected behavior

As `adapter-transformers` is a drop-in replacement of the `transformers` library, I expect random numbers (and randomly initialized layers added) to be the same between the two libraries, for a given random seed.

My guess is that `adapter-transformers` v3.1.0 code is drawing extra random values compared to `adapter-transformers` v3.0.0 or `transformers` v4.23.1.

**Comment by calpt** [comment]
@couturierc Thanks for investigating this.

As explanation: The discrepancy was introduced in v3.1.0 due to the integration of LoRA. In the GPT-2 implementation, `Conv1D` layers were replaced by `LoRALinear` layers to accomodate the new LoRA adapters. While both layers provide the same functionality, they derive from different classes (`Conv1D` is a custom class by HF Transformers while `LoRALinear` derives from PyTorch's `Linear` layer). Due to this fact, there is a slight difference in the way bias parameters are initialized by default at instantiation time. (In practice, all weights will be initialized a second time by the respective model class in `init_weights()`, overwriting the default initialization).

I'm providing a fix to restore full compatibility in #525.
